### ISSUE TITLE
support fetching dependencies with the file:// protocol scheme

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#17364]
This should work:

```
zig fetch file://C:/Code/myproject/
```

and of course the equivalent build.zig.zon snippet:

```
    .url = "file://C:/Code/myproject/"
```

Support `file://` just like any other URL scheme.

URLs already can be private or require a proxy, or blocked in some countries. File paths have the exact same properties. There is no reason to not support this kind of URL.

If someone has a security concern about this and wants to be taken seriously then please explain the threat model and let's analyze it.

The previous issue for this feature was #14339 however, that was two separate use cases, and the resulting enhancement that came out of that was paths relative to the build root which are exempt from hashing. The use case for supporting `file:///` still remains and is still planned.

**Comment by nektro** [comment]
small nit: it should be the `file://` protocol, it commonly has 3 slashes because on non-windows systems absolute paths start with `/`

**Comment by andrewrk** [comment]
Landed in f7bc55c0136b91805bd046a8cc8ea745d7e7567d.
